qutebrowser/mainwindow/statusbar/page_ram_usage.py

```diff
@@ -0,0 +1,104 @@
+"""Page RAM usage displayed in the statusbar."""
+
+import platform
+from typing import Optional
+from qutebrowser.browser import browsertab
+from qutebrowser.misc import throttle
+from qutebrowser.qt.core import Qt
+from qutebrowser.qt.widgets import QWidget
+from qutebrowser.qt.gui import QHideEvent, QShowEvent
+
+from qutebrowser.mainwindow.statusbar import textbase
+from qutebrowser.utils import usertypes
+from qutebrowser.utils import log
+
+
+class PageRamUsage(textbase.TextBase):
+    """Show memory usage of the renderer process."""
+
+    @staticmethod
+    def __mem_usage_darwin(_: int) -> str:
+        # pid
+        raise NotImplementedError()
+
+    @staticmethod
+    def __mem_usage_windows(_: int) -> str:
+        # pid
+        raise NotImplementedError()
+
+    @staticmethod
+    def __mem_usage_linux(pid: int) -> str:
+        _proc_status = f"/proc/{pid}/status"
+        _scale = {"kB": 1024.0, "mB": 1024.0*1024.0, "KB": 1024.0, "MB": 1024.0*1024.0}
+
+        def _mem_bytes(vm_key: str) -> float:
+            """Private."""
+            # get pseudo file  /proc/<pid>/status
+            try:
+                with open(_proc_status, encoding="utf-8") as t:
+                    v = t.read()
+            except Exception:
+                log.statusbar.exception(f"failed to open {_proc_status} file")
+                return 0.0  # non-Linux?
+            # get VmKey line e.g. "VmRSS:  9999  kB\n ..."
+            i = v.index(vm_key)
+            parts = v[i:].split(None, 3)  # whitespace
+            if len(v) < 3:
+                return 0.0  # invalid format?
+            # convert Vm value to bytes
+            return float(parts[1]) * _scale[parts[2]]
+
+        def resident(since: float = 0.0) -> float:
+            """Return resident memory usage in bytes."""
+            return _mem_bytes("VmRSS:") - since
+
+        mem = resident() / _scale["mB"]
+        return "%.2f" % mem
+
+    UPDATE_DELAY = 1000  # ms
+
+    def __init__(self, parent: Optional[QWidget] = None) -> None:
+        super().__init__(parent, elidemode=Qt.TextElideMode.ElideNone)
+        self._set_text = throttle.Throttle(self.setText, 100, parent=self)
+        self.timer = usertypes.Timer(self)
+        self.timer.timeout.connect(self._show_ram_usage)
+        self.pid = 0
+
+    def _show_ram_usage(self) -> None:
```

Comment:
Instead of reimplementing the wheel by reading `/proc`, I think here it'd make sense to add [`psutil`](https://psutil.readthedocs.io/) as an optional dependency. You'll need some more adjustments around the docs and `qutebrowser/utils/version.py`, but seems like it'd be worth it.

Reply:
Can I use it? That would be helpful; I just was not sure if I can add dependencies or not.

Reply:
@The-Compiler  Noob's question. To add `psutil` into dependencies I do:
1. Add it into misc/requirements/requirements-qutebrowser.txt-raw . 
2. Run scripts/dev/recompile_requirements 

That's probably it. Or there is a better and simpler way?

Reply:
@Lezh1k I think that's it, and add it alongside the other optional dependancies like Colorama. From memory you can run `scripts/dev/recompile_requirements.py qutebrowser` to just recompile that files. You probably want to add psutils to the test requirements files too though (again, copy colorama).

And in case you missed it, I think the `qutebrowser/utils/version.py` reference is regarding [this bit](https://github.com/qutebrowser/qutebrowser/blob/4a1a30b/qutebrowser/utils/version.py#L385) where we report on whether optional dependancies are installed, and their version in the `:version` page.